python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -986,31 +699,24 @@ def read_multiple_files(paths, columns=None, use_threads=True, **kwargs):
     t = pa.Table.from_pandas(bad_apple)
     _write_table(t, bad_apple_path)
 
-    if not use_legacy_dataset:
-        # TODO(dataset) Dataset API skips bad files
-        return
+    # TODO(dataset) Dataset API skips bad files
```

Comment:
Do we know if there is an issue about this?

Reply:
I haven't found one. I am trying to figure out what should be tested and will update it so that skipping bad files in Dataset API is included in it.

Reply:
Can also open an issue and resolve this TODO in a separate PR.

Reply:
Yes, I think that's good, let's further figure this out in a follow-up issue

Reply:
Opened: https://github.com/apache/arrow/issues/39338